packages/manager/apps/container/src/container/nav-reshuffle/sidebar/SidebarLink.tsx

```diff
@@ -56,9 +62,12 @@ const SidebarLink: React.FC<ComponentProps<SidebarLinkProps>> = ({
       data-testid={id}
       role="button"
     >
-      <span className='flex gap-2 align-items-center'>
-        <SvgIconWrapper name={node.svgIcon} height={32} width={32} className='p-1 fill-white block' />
-        {!isShortText && <span>{t(node.translation)}</span>}
+      <span className='flex gap-2 align-items-center w-full whitespace-nowrap'>
+        <SvgIconWrapper name={node.svgIcon} height={32} width={32} className={`p-1 fill-white transition-all duration-200O shrink-0 ${isLoading && isShortText ? 'hidden' : 'block'}`} />
+        {isShortText &&
+          <OsdsSpinner className={`h-[32px] w-[32px] transition-all duration-200O shrink-0 ${isLoading ? 'block' : 'hidden'}`} size={ODS_SPINNER_SIZE.sm} contrasted />
+        }
```

Comment:
Can't we make the display conditional in another way ? Because here we will add the SvgIconWrapper and Spinner in the dom just to hide them
```suggestion
        {isLoading && isShortText && (
          <>
            <SvgIconWrapper name={node.svgIcon} height={32} width={32} className={`p-1 fill-white transition-all duration-200O shrink-0 block`} />
            <OsdsSpinner className={`h-[32px] w-[32px] transition-all duration-200O shrink-0 block`} size={ODS_SPINNER_SIZE.sm} contrasted />
          </>
        )}
```